v2/samples/kusto/v1api20240413/v1api20240413_principalassignment.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: kusto.azure.com/v1api20240413
+kind: PrincipalAssignment
+metadata:
+  name: aso-sample-principalassignment
+  namespace: default
+spec:
+  owner:
+    name: aso-sample-database
+  principalId: "james.judd@aveva.com"
```

Comment:
I had to go look up the docs for this.

Usually, a `principalId` is strictly a GUID, but in this case it's much more flexible:

>The principal ID assigned to the cluster principal. It can be a user email, application ID, or security group name.

Nice.